Microsoft.Toolkit.Mvvm/ComponentModel/ObservableRecipient.cs

```diff
@@ -15,27 +15,28 @@
 namespace Microsoft.Toolkit.Mvvm.ComponentModel
 {
     /// <summary>
-    /// A base class for viewmodels.
+    /// A base class for observable objects that also acts as recipients for messages. This class is an extension of
+    /// <see cref="ObservableObject"/> which also provides built-in support to use the <see cref="IMessenger"/> type.
     /// </summary>
-    public abstract class ViewModelBase : ObservableObject
+    public abstract class ObservableRecipient : ObservableObject
```

Comment:
What's the reasoning behind this change?
"ViewModelBase" is a widely used name and its purpose is implied from its name.
The name "ObservableRecipient" makes me think it is something that will receive "observables".
This new name seems to be very much focused on the ability to receive messages. Will there be an equivalent object that can be used as a general-purpose base class for anyone not using messages? Or should they just use ObservableObject directly? If this is intended to be an alternative to ObservableObject that also broadcasts messages when properties are changed, why not a name that reflects this?
This class doesn't _receive_ anything, but rather it sends/broadcasts messages. Were names like "BroadcastingObservableObject" also considered? Why was "ObservableRecipient" deemed the most appropriate name to use?

Reply:
Hey Matt, that's a good question, and something I plan to properly document in the docs too! 😄

I made these changes after going through all the feedbacks from @deanchalk and discussing this with @michael-hawker. Basically, the idea is that the `ViewModelBase` name was itself confusing (and rightly so) for many users: the name implied it was just a "general purpose" base for viewmodels, when in fact that wasn't really the case. All the basic viewmodel functionality was already provided just by `ObservableObject`, and `ViewModelBase` was solely focused on doing proper integration with the `IMessenger` type. As a result, users might've felt "forced" to use the messenger in all cases (which wasn't the intent here), or could've just looked at that `Messenger` property thinking "I don't need this, why is it here?".

As a result, I've now renamed that `ObservableRecipient`, which also is more consistent with the general naming scheme: "Observable" like the other base class, and "recipient" which is the term used in all the messenger APIs. This class is now explicitly a more specialized version of `ObservableObject` that is meant to be used to support `IMessenger`, providing both new overloads for the `Set` method that integrate with it, as well as adding automatic setup for messages, if needed. And also, with a couple of constructors to use with DI, either automatically grabbing the shared `IMessenger`, or letting users inject their own from a DI service provider, so offer maximum flexibility.

> Will there be an equivalent object that can be used as a general-purpose base class for anyone not using messages? Or should they just use ObservableObject directly?

Yes, people not interested in using messages should just inherit from `ObservableObject`, that class already provides all the basic functionalities to support the `INotifyProperty[Changed|Changing]` interface(s).

> This class doesn't receive anything, but rather it sends/broadcasts messages. Were names like "BroadcastingObservableObject" also considered? Why was "ObservableRecipient" deemed the most appropriate name to use?

This is not correct, this class does act as a recipient. In fact, in many cases (eg. in my experience) it often acts more as a recipient than a broadcaster (hence the name too). You can see that the base implementation of `OnActivated` (as suggested by Michael) automatically registers all the declared message handlers (through the `IRecipient<TMessage>` interface), and users also have the ability to fine tune the subscription more by overriding that method. The `OnDeactivated` method also automatically unregisters all the message handlers, and these two are invoked by the setter for `IsActive`. In general, this class offers ready to use support for receiving messages, first and foremost. The ability to also broadcast messages is just a bonus, also because sending messages requires less setup than receiving one.

Hope this helps!

Reply:
The terminology and lack of documentation certainly have the potential to be very confusing. As is, this has the potential to receive messages but doesn't (as far as I can tell) receive anything itself. (yet?)